cms/djangoapps/contentstore/management/commands/reindex_course.py

```diff
@@ -0,0 +1,111 @@
+""" Management command to update courses' search index """
+import logging
+from django.core.management import BaseCommand, CommandError
+from optparse import make_option
+from textwrap import dedent
+
+from contentstore.courseware_index import CoursewareSearchIndexer
+from search.search_engine_base import SearchEngine
+from elasticsearch import exceptions
+
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+from opaque_keys.edx.locator import CourseLocator
+
+from .prompt import query_yes_no
+
+from xmodule.modulestore.django import modulestore
+
+
+class Command(BaseCommand):
+    """
+    Command to re-index courses
+
+    Examples:
+
+        ./manage.py reindex_course <course_id_1> <course_id_2> - reindexes courses with keys course_id_1 and course_id_2
+        ./manage.py reindex_course --all - reindexes all available courses
+        ./manage.py reindex_course --setup - reindexes all courses for devstack setup
+    """
+    help = dedent(__doc__)
+
+    can_import_settings = True
+
+    args = "<course_id course_id ...>"
+
+    all_option = make_option('--all',
+                             action='store_true',
+                             dest='all',
+                             default=False,
+                             help='Reindex all courses')
+
+    setup_option = make_option('--setup',
+                               action='store_true',
+                               dest='setup',
+                               default=False,
+                               help='Reindex all courses on developers stack setup')
+
+    option_list = BaseCommand.option_list + (all_option, setup_option)
+
+    CONFIRMATION_PROMPT = u"Re-indexing all courses might be a time consuming operation. Do you want to continue?"
+
+    def _parse_course_key(self, raw_value):
+        """ Parses course key from string """
+        try:
+            result = CourseKey.from_string(raw_value)
+        except InvalidKeyError:
+            result = SlashSeparatedCourseKey.from_deprecated_string(raw_value)
```

Comment:
This block can except two times, throwing exception again in except block. I think this was properly addressed in delete_course command. 
Also, since this pattern is used multiple times across platform (especially commands), we might remove duplicated code and simplify course_key parsing changes in future.